src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1194,6 +1195,32 @@ public function getRouteKeyName()
     }
 
     /**
+     * Get the translation name for the model.
+     *
+     * @return string
+     */
+    public function getTranslationName()
+    {
+        $key = $this->getTranslationNameKey();
+
+        if (Lang::has($key)) {
+            return Lang::get($key);
+        }
+
+        return Str::snake(class_basename($this), ' ');
+    }
+
+    /**
+     * Get the translation name key for the model.
+     *
+     * @return string
+     */
+    public function getTranslationNameKey()
+    {
+        return 'models.'.Str::snake(class_basename($this));
+    }
```

Comment:
Route key is fine because it doesn't introduce usage of any "facade", and none of the current usage do it right now.